pmd-java/src/main/java/net/sourceforge/pmd/lang/java/qname/QualifiedNameFactory.java

```diff
@@ -108,7 +108,7 @@ public static JavaTypeQualifiedName ofClass(Class<?> clazz) {
             name = '.' + name; // unnamed package, marked by a full stop. See ofString's format below
         }
 
-        return (JavaTypeQualifiedName) ofString(name);
+        return ((JavaTypeQualifiedName) ofString(name)).withClassLoader(clazz.getClassLoader());
```

Comment:
this can fail too and not be the expected classloader. The auxclasspath refers to the auxclasspath entries, and have the PMD classloader as parent (which allows it to load the JRE classes). I'm not certain there actually exists a scenario where this could matter, moving from a JRE class back to our domain so maybe we don't even care...